core/auth/src/test/java/software/amazon/awssdk/auth/token/credentials/ProfileTokenProviderTest.java

```diff
@@ -73,6 +74,36 @@ public void compatibleProfileSettings_callsLoader() {
         assertThatThrownBy(provider::resolveToken).hasMessageContaining("does not have sso_session property");
     }
 
+    @Test
+    void resolveToken_profileFileSupplier_suppliesObjectPerCall() {
+        ProfileFile file1 = profileFile("[profile sso]\n"
+                                       + "aws_access_key_id = defaultAccessKey\n"
+                                       + "aws_secret_access_key = defaultSecretAccessKey\n"
+                                       + "sso_session = xyz");
+        ProfileFile file2 = profileFile("[profile sso]\n"
+                                       + "aws_access_key_id = modifiedAccessKey\n"
+                                       + "aws_secret_access_key = modifiedSecretAccessKey\n"
+                                       + "sso_session = xyz");
+        Supplier<ProfileFile> supplier = Mockito.mock(Supplier.class);
+
+        ProfileTokenProvider provider =
+            ProfileTokenProvider.builder().profileFile(supplier).profileName("sso").build();
+
+        try {
+            Mockito.when(supplier.get()).thenReturn(file1);
+            provider.resolveToken();
+        } catch (IllegalArgumentException e) {
```

Comment:
Yes, every functional test I reviewed (at least in this class) involving SSO cannot resolve and throw an exception. Since I wanted to verify two invocations, I could only think of swallowing the first exception to continue processing.

Reply:
Gotcha, can we add a comment in catch block?

Reply:
I just realized i could have asserted the exception instead. Think I should make the change?

```
Mockito.when(...);
assertThatThrownBy(provider::resolveToken)....;

Mockito.verify(supplier, ...);
```

Reply:
Yeah, I was going to suggest that. Sounds good to me.

Reply:
Restoring comment deleted by mistake. Will add assertions to better handle the exceptions.

```
Mockito.when(...);
assertThatThrownBy(provider::resolveToken)....;

Mockito.verify(supplier, ...);
```